build.rs

```diff
@@ -339,7 +340,10 @@ fn check_features(infos: &Vec<(&'static str, Option<&'static str>, &'static str)
 fn main() {
 	if env::var("CARGO_FEATURE_BUILD").is_ok() {
 		println!("cargo:rustc-link-search=native={}", search().join("lib").to_string_lossy());
-		println!("cargo:rustc-link-lib=z");
+
+		if env::var("CARGO_FEATURE_ZLIB").is_ok() && cfg!(target_os = "linux") {
```

Comment:
It does but the Windows dependency is handled in a much different way according to what I read on the subject.  Leaving open to any OS seemed to be more of a problem than restricting until someone verified it works on other OS.